### ISSUE TITLE
test against stock version

### ISSUE THREAD

**Issue by steven-murray** [steven-murray/py21cmmc#6]
It would be really really good to get the Master's judgment on how well the code performs, relative to the standard version. Any ideas, @BradGreig?

**Comment by BradGreig** [comment]
@steven-murray what does the standard version refer to? Are you thinking of code timing? You can run a single instance on both versions and compare the total computational time taken. Harder to compare an actual MCMC run though as the computational time is dependent on the astrophysical parameters

**Comment by steven-murray** [comment]
Standard version meaning the exact code you sent (called "March21st"). I think timing is OK -- I have tested to make sure that this is roughly conserved. I also think that I get the same answer out from both codes, in the most obvious use-case. But I do not have the same level of experience using the code, and all the corner cases it might throw up. I would like this code to be able to deal with those cases with as much or more robustness as the original. This will especially relate to the various flag options, as well as how the MCMC is run (so not so much making sure the MCMC converges to the same answer, but making sure you can *do* the same things with it).

**Comment by BradGreig** [comment]
Other than running the code for the various flag options, I am not sure what else you can actually do. The minimum test I do is to set the MCMC to only do one iteration (burnin=0) and select the number of threads that corresponds to the half the total number of walkers (walkersratio=2 * number of astrophysical parameters).

This allows you to test the MCMC, and the multiprocessing while also testing the outputs. You could then save all outputs and compare results. To do this, you might want to force the same astrophysical parameters to be sampled each time.

To test/guard against all corner cases is becoming difficult given the total number of combinations available with the flags.

**Comment by steven-murray** [comment]
I agree that it will be difficult to test all combinations. I guess I would like it to "work" (i.e., at least _intend_ on doing the right thing) for at least the kinds of cases that you would have usually run. 

Eg. there are parameters like "KEEP_ALL_DATA" in the MCMC options. I am not exactly sure what they do -- what does "ALL_DATA" refer to? So in simplifying the code, I may have inadvertantly changed this behaviour. I feel like if (when you get the time, of course) you run very minimal versions of runs you would have run with your code, you'll quickly find these kinds of bugs where the behaviour is changed in the new version, and we will be able to fix them up.

**Comment by BradGreig** [comment]
"KEEP_ALL_DATA" provided the user the option to store all Walker* files, the 21cmPS file, global averaged histories and the optical depth data. Basically, it would switch the MCMC from deleting this information to storing the information in a directory.

I can do this testing once the finalised structure of 21CMMC is in place.

**Comment by steven-murray** [comment]
Yup cool. This issue is really just a placeholder until that time to keep my mind straight :-)
